aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -28,32 +28,51 @@
 
 #include <map>
 
+namespace at {
+namespace native {
+
 // NOTE:
-// YOU ARE NOT OBLIGED TO USE THESE MACROS
+// YOU ARE NOT OBLIGED TO USE THESE HELPERS
 // If you're writing something more specialized, please don't try to make them
 // work for your case, but just write something new instead.
 
-namespace at {
-namespace native {
+// Helper functions that reduce redundant code in implementing the most typical kind of unary operators. This allows
+// some preprocessing that are unique to some operators (more is forsee-able in the future) and is more flexible and
+// elegant than defining a flat fat macro that implements everything.
+template <typename Stub>
+static inline Tensor& unary_op_out_impl(Tensor& result, const Tensor& self, Stub& stub) {
```

Comment:
nit: i'd rename this unary_op_impl_out, keeping "_" and "out" at the end makes it easier to read.